src/lib/chips/chip-list.ts

```diff
@@ -3,17 +3,31 @@ import {
   ChangeDetectionStrategy,
   Component,
   ContentChildren,
+  Directive,
   Input,
   QueryList,
   ViewEncapsulation,
   OnDestroy,
+  ElementRef,
+  Renderer2,
 } from '@angular/core';
 
 import {MdChip} from './chip';
 import {FocusKeyManager} from '../core/a11y/focus-key-manager';
 import {coerceBooleanProperty} from '../core/coercion/boolean-property';
-import {SPACE, LEFT_ARROW, RIGHT_ARROW, TAB} from '../core/keyboard/keycodes';
 import {Subscription} from 'rxjs/Subscription';
+import {
+  LEFT_ARROW, RIGHT_ARROW, SPACE, BACKSPACE, DELETE, UP_ARROW, DOWN_ARROW
+} from '../core/keyboard/keycodes';
+import {Dir} from '../core/rtl/dir';
+
+@Directive({
+  selector: '[mdChipListContainer], [matChipListContainer]',
```

Comment:
My ideal would be to first do that work we've talked about for making `md-input-container` work with other types of inputs besides just native inputs. We could then have `md-chip-list` implement the interface and act as an input (possibly a readonly one if the user doesn't actually plop a native input inside of it). The reason I like this is that the `md-chip-list ` is then responsible for things like deciding when its empty, it can take into account both committed chips and uncommitted chips in the form of input that is currently being entered. In the end it would wind up looking similar to what tina suggested:

```html
<md-input-container>
  <md-chip-list>
    <md-chip>Chip 1</md-chip>
    <input/>
  </md-chip-list>
</md-input-container>
```
The key difference here though is that the native `<input>` is fully managed by the `md-chips-list`, the `md-input-container` knows nothing about the native input and instead interacts with the `md-chip-list` through some `MdInput` interface that the chips list implements.

If we don't want to do all this work now, then maybe we can make it look like tina's suggestion so its less of a change when we can eventually do it this way?